src/mame/drivers/terco.cpp

```diff
@@ -0,0 +1,103 @@
+// license:BSD-3-Clause
+// copyright-holders:Joakim Larsson Edstrom
+/*
+ * The 4490 CNC Mill Control unit
+ *__________________________________________________________________________________________________________
+ *                                                                                                          |
+ *__________________________________________________________________________________________________________|
+ *
+ * History of Terco
+ *------------------
+ * 
+ * Misc links about the boards supported by this driver.
+ *-----------------------------------------------------
+ *
+ *  TODO:
+ * ------
+ *  - Add PCB layouts     
+ *  - Keyboard            
+ *  - Display/CRT         
+ *  - Clickable Artwork   
+ *  - Cassette i/f
+ *  - Expansion bus
+ */
+
+#include "emu.h"
+#include "cpu/m6800/m6800.h"
+#include "machine/6821pia.h"
+//#include "machine/6850acia.h"
+
+#define LOG_SETUP   (1U <<  1)
+#define LOG_READ    (1U <<  2)
+#define LOG_BCD     (1U <<  3)
+
+//#define VERBOSE (LOG_BCD|LOG_SETUP)
+//#define LOG_OUTPUT_STREAM std::cout
+
+#include "logmacro.h"
+
+#define LOGSETUP(...) LOGMASKED(LOG_SETUP,   __VA_ARGS__)
+#define LOGR(...)     LOGMASKED(LOG_READ,    __VA_ARGS__)
+#define LOGBCD(...)   LOGMASKED(LOG_BCD,     __VA_ARGS__)
+
+#ifdef _MSC_VER
+#define FUNCNAME __func__
+#else
+#define FUNCNAME __PRETTY_FUNCTION__
+#endif
+
+/* Terco CNC Control Station 4490 */
+class t4490_state : public driver_device
+{
+public:
+	t4490_state(const machine_config &mconfig, device_type type, const char *tag)
+	: driver_device(mconfig, type, tag)
+	,m_maincpu(*this, "maincpu")
+	,m_pia1(*this, "pia1")
+	,m_pia2(*this, "pia2")
+	  //	,m_acia(*this, "acia")
+	{ }
+	required_device<m6800_cpu_device> m_maincpu;
+	virtual void machine_reset() override { m_maincpu->reset(); LOG("--->%s()\n", FUNCNAME); };
```

Comment:
Overridden protected member should be protected, and if `m_maincpu` needs to be public, you probably have design issues.

Reply:
Oops I missed that, the driver has been lying around in my repo for a long time, just wanted to get into mainstream, that override is a remain from some experiments I suspect. Will fix.